reconcile/test/test_checkpoint.py

```diff
@@ -0,0 +1,174 @@
+from http import HTTPStatus
+
+import pytest
+import reconcile.checkpoint as sut
+import requests
+
+
+@pytest.fixture
+def valid_app():
+    """How a valid application looks like."""
+    return {
+        "sopsUrl": "https://www.redhat.com/sops",
+        "architectureDocument": "https://www.redhat.com/arch",
+        "grafanaUrl": "https://www.redhat.com/graf",
+        "serviceOwners": [{"name": "A Name", "email": "aname@adomain.com"}],
+    }
+
+
+@pytest.fixture
+def valid_owner():
+    """How a valid owner looks like."""
+    return {"name": "A Name", "email": "a.name@redhat.com"}
+
+
+@pytest.fixture
+def invalid_owners():
+    """List the ways in which an owner can be invalid."""
+    return [
+        {"name": "A Name", "email": None},
+        {"name": "A Name", "email": "domainless"},
+        {"name": "A Name", "email": "@name.less"},
+        {"name": None, "email": "a-name@redhat.com"},
+    ]
+
+
+def test_valid_owner(valid_owner) -> None:
+    """Confirm that the valid owner is recognized as such."""
+    assert sut.valid_owners([valid_owner])
+
+
+def test_invalid_owners(invalid_owners):
+    """Confirm that the invalid owners are flagged."""
+    for o in invalid_owners:
```

Comment:
Might make more sense to use parameterization here: https://docs.pytest.org/en/6.2.x/parametrize.html

If you parametrize the test, they'll show up in the results as distinct tests, and they'll all run instead of short-circuiting. You could just make this `INVALID_OWNERS` and even `return INVALID_OWNERS` if you really needed the fixture too for some reason.